src/core/radio/radio_platform.cpp

```diff
@@ -250,6 +250,13 @@ OT_TOOL_WEAK uint32_t otPlatRadioGetBusSpeed(otInstance *aInstance)
     return 0;
 }
 
+#if OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE
+OT_TOOL_WEAK otError otPlatRadioResetCsl(otInstance *aInstance)
+{
+    return otPlatRadioEnableCsl(aInstance, 0, Mac::kShortAddrInvalid, nullptr);
```

Comment:
Is there any different behavior that a platform implementation would do in `otPlatRadioResetCsl()` implementation that would not do in  `otPlatRadioEnableCsl(0, Mac::kShortAddrInvalid, nullptr)`?

>  nulltpr is the main issue for us, as on Radio Driver layer, Ext Address is checked if it is not NULL to prevent handling pointer incorrectly. For current implementation we would need to consider a special case, which would accept NULL pointer and it doesn't sound very secure.

If the goal is that the `otPlatRadioEnableCsl()` would no longer support disabling and the `(0, Mac::kShortAddrInvalid, nullptr)` would not be handled correctly (e.g. implemention wont check for `nullptr` ext-address ), it may itself cause issues/bugs (since it is part of current documented behavior of this API).

> Also, EnableCsl and otPlatRadioEnableCsl in documentation don't specify the case of resetting the values, they only specify setting them (with no explanation of the special case described before in particular). 

I personally favor having fewer APIs. Makes it easier for platforms to implement. So one API to enable or disable CLS on radio platform sounds better to me than two APIs for this.

So how about we update the documentation of the `otPlatRadioEnableCsl()` to make it clear that when disabling (with zero `aCslPersiod`)  the ext-address can be `nullptr` (and should be ignored by platform).

Reply:
In case of nordic's implementation in Zephyr,`otPlatRadioEnableCsl` when CSL period is set to 0, Removes the address of a peer node for which the ACK data is set from the pending bit list. With `otPlatRadioResetCsl`, it makes sure to clear all addresses  from the ACK data list.
> Is there any different behavior that a platform implementation would do in `otPlatRadioResetCsl()` implementation that would not do in `otPlatRadioEnableCsl(0, Mac::kShortAddrInvalid, nullptr)`?
 

`otPlatRadioEnableCsl` should still support disabling and enabling CSL. It is now called only in `SubMac::UpdateCsl(uint16_t aPeriod, uint8_t aChannel, otShortAddress aShortAddr, const otExtAddress *aExtAddr)` (apart from the other place I'll explain later in this comment). From inspecting the code, I can see that it is only called in sub_mac:
```
bool SubMac::UpdateCsl(uint16_t aPeriod, uint8_t aChannel, otShortAddress aShortAddr, const otExtAddress *aExtAddr)
{
    bool diffPeriod  = aPeriod != mCslPeriod;
    bool diffChannel = aChannel != mCslChannel;
    bool diffPeer    = aShortAddr != mCslPeerShort;
    bool retval      = diffPeriod || diffChannel || diffPeer;

    VerifyOrExit(retval);
    mCslChannel = aChannel;

    VerifyOrExit(diffPeriod || diffPeer);
    mCslPeriod    = aPeriod;
    mCslPeerShort = aShortAddr;
    IgnoreError(Get<Radio>().EnableCsl(aPeriod, aShortAddr, aExtAddr));

    mCslTimer.Stop();
    if (mCslPeriod > 0)
    {
        mCslSampleTime = TimeMicro(static_cast<uint32_t>(otPlatRadioGetNow(&GetInstance())));
        mIsCslSampling = false;
        HandleCslTimer();
    }

exit:
    return retval;
}
```

and given function is called in `mac.cpp`:
```
void Mac::UpdateCsl(void)
{
    uint16_t period  = IsCslEnabled() ? GetCslPeriod() : 0;
    uint8_t  channel = GetCslChannel() ? GetCslChannel() : mRadioChannel;

    if (mLinks.UpdateCsl(period, channel, Get<Mle::Mle>().GetParent().GetRloc16(),
                         &Get<Mle::Mle>().GetParent().GetExtAddress()))
    {
        if (Get<Mle::Mle>().IsChild())
        {
            Get<DataPollSender>().RecalculatePollPeriod();

            if (period != 0)
            {
                Get<Mle::Mle>().ScheduleChildUpdateRequest();
            }
        }

        UpdateIdleMode();
    }
}
```
I cannot see if there could be a situation when `&Get<Mle::Mle>().GetParent().GetExtAddress()` could become a `nullptr`, but there is also an `IgnoreError` guard around `EnableCsl`.

> If the goal is that the `otPlatRadioEnableCsl()` would no longer support disabling and the `(0, Mac::kShortAddrInvalid, nullptr)` would not be handled correctly (e.g. implemention wont check for `nullptr` ext-address ), it may itself cause issues/bugs (since it is part of current documented behavior of this API).

I agree, but the place where I put `ResetCsl` call is a bit of a different case. It's point is not to just disable CSL, but to reset stack on RCP reset (implemented here: https://github.com/openthread/openthread/pull/6645/files#diff-594785f402ea6b1c22fa169db33d2335ef80223cd1084537a5627225f86c4f2fR44) . In this case, I think we should differentiate resetting CSL parameters from just disabling the CSL?

> I personally favor having fewer APIs. Makes it easier for platforms to implement. So one API to enable or disable CLS on radio platform sounds better to me than two APIs for this.
> 
> So how about we update the documentation of the `otPlatRadioEnableCsl()` to make it clear that when disabling (with zero `aCslPersiod`) the ext-address can be `nullptr` (and should be ignored by platform).